### PR TITLE
feat: port check storage command into Rust

### LINKED ISSUE #1189: Convert check storage command from Python to Rust

### PR DESCRIPTION
**Pull request by bbangert**
Co-authored by: Phil Jenvey <pjenvey@underboss.org>

Some components of CheckStorage are left in Python as they're used
by the delete tests.

Closes #1189

### CODE REVIEW

File: autopush_rs/src/util/ddb_helpers.rs [hunk 1]
```diff
@@ -102,3 +129,479 @@ macro_rules! ddb_item {
...
+        static ref RE: RegexSet = RegexSet::new(&[
+            r"^01:\S+:\S+$",
+            r"^02:\d+:\S+$",
+            r"^\S{3,}:\S+$",
+        ]).unwrap();
+    }
+    if !RE.is_match(key) {
+        return Err("Invalid chidmessageid".into()).into();
+    }
+    if key.starts_with("01:") {
```
Comment (jrconlin) [inline autopush_rs/src/util/ddb_helpers.rs]: part of me wonders if this should be a match...

Comment (bbangert) [reply autopush_rs/src/util/ddb_helpers.rs]: I don't think strings can be destructured to partial match right now.

File: autopush_rs/src/util/ddb_helpers.rs [hunk 2]
```diff
@@ -102,3 +129,479 @@ macro_rules! ddb_item {
...
+                    limit: Some(limit as i64),
+                    ..Default::default()
+                })
+            },
+            |err: &QueryError| matches!(err, &QueryError::ProvisionedThroughputExceeded(_)),
+        ).map_err(|_| "Error fetching messages".into());
+        let response = response.and_then(|data| {
+            let notifs: Option<Vec<Notification>> = data.items.and_then(|items| {
+                debug!("Got response of: {:?}", items);
+                // TODO: Capture translation errors and report them as we shouldn't have corrupt data
```
[the remark this answers is not in the document]
Comment (bbangert) [reply autopush_rs/src/util/ddb_helpers.rs]: Yea, I was saving this for after we go to Sentry 0.3 so we can easily log them out to Sentry.

File: autopush_rs/src/util/ddb_helpers.rs [hunk 3]
```diff
@@ -102,3 +129,479 @@ macro_rules! ddb_item {
...
+            let mut timestamp = None;
+            if let Some(ref messages) = notifs {
+                if messages.len() == 0 {
+                    return Ok(Default::default());
+                }
+                timestamp = messages.iter().filter_map(|m| m.sortkey_timestamp).max();
+            }
+            Ok(FetchMessageResponse {
+                timestamp,
+                messages: notifs,
```
[the remark this answers is not in the document]
Comment (bbangert) [reply autopush_rs/src/util/ddb_helpers.rs]: We'll have to change how the caller and responder work to do that as well, which is a bit more of a change than just porting the Python side.

**Pull request merged**